### ISSUE TITLE
increasing character limit beyond 140

### ISSUE THREAD

**Issue by geffsmith** [ggerganov/ggwave#35]
hello..

Firstly amazing tool..incredibly fast and very responsive. Has been trying to increase the 140 character limit on variable length string(very large say 10,000 characters). But stuck at segmentation fault at various stages. Any guidance will be greatly appreciated.

thanks

**Comment by ggerganov** [comment]
Hi, check out this discussion for more info on this topic:

https://github.com/ggerganov/ggwave/discussions/34

I will think about extending the protocol for long transmissions, but the biggest problem is how to make the communication reliable for such long periods.

Just curious (if not a secret) - for what application do you need to transmit such long messages?

**Comment by geffsmith** [comment]
hello,

Thanks for the quick reply. That explains it in very detail. Was trying to send a file using this library.

**Comment by geekmush** [comment]
Just a data point ... to address our needs of larger messages, I just took the python binding and used it to add support for fragmenting the messages.  It would probably be better to implement in the core C library, but I could do it in the python faster (or else I'd offer a PR).

Let me know if this interests you as a reference for the idea.

**Comment by ggerganov** [comment]
Yes, splitting the message into smaller chunks is always an option.
I will think about adding it as an option in the core library. But I am not sure about the best way to do that.
- Does the transmitter first broadcast the expected number of chunks?
- Does the receiver confirm when they've received a given chunk successfully?
- Does the transmitter re-send a chunk of the receiver didn't received it successfully?

Will think about it
